database/src/lib.rs

```diff
@@ -1,520 +1,110 @@
-//         block_height: u64,
-//     ) -> anyhow::Result<()> {
-//         let transaction_details = transaction
-//             .try_to_vec()
-//             .expect("Failed to borsh-serialize the Transaction");
-//          Self::execute_prepared_query(
-//             &self.scylla_session,
-//             &self.add_transaction,
-//             (
-//                 transaction.transaction.hash.to_string(),
-//                 num_bigint::BigInt::from(block_height),
-//                 transaction.transaction.signer_id.to_string(),
-//                 &transaction_details,
-//             )
-//         ).await?;
-//         Ok(())
-//     }
-// }
-//
-// Usage:
-// let scylla_db_client = MyScyllaManager::new(
-//         scylla_url,
-//         scylla_keyspace,
-//         scylla_user,
-//         scylla_password,
-//         Some(keepalive_interval),
-// ).await?,
-
-use scylla::prepared_statement::PreparedStatement;
-use scylla::retry_policy::{QueryInfo, RetryDecision};
-use scylla::transport::errors::QueryError;
-
-#[derive(Debug)]
-pub struct CustomDBRetryPolicy {
-    max_retry: u8,
-    strict_mode: bool,
-}
-
-impl CustomDBRetryPolicy {
-    pub fn new(max_retry: u8, strict_mode: bool) -> CustomDBRetryPolicy {
-        CustomDBRetryPolicy {
-            max_retry,
-            strict_mode,
-        }
-    }
-}
-
-impl Default for CustomDBRetryPolicy {
-    /// By default using `strict_mode`
-    fn default() -> CustomDBRetryPolicy {
-        CustomDBRetryPolicy::new(0, true)
-    }
+mod base;
+
+use crate::base::BaseDbManager;
+pub use crate::base::ReaderDbManager;
+pub use crate::base::StateIndexerDbManager;
+pub use crate::base::TxIndexerDbManager;
+#[cfg(feature = "scylla_db")]
+mod scylladb;
+#[cfg(feature = "scylla_db")]
+pub(crate) use crate::scylladb::base::ScyllaStorageManager;
+
+pub mod primitives;
+
+/// Additional database options for scylla database
+///
+/// `preferred_dc: Option<String>`,
+/// ScyllaDB preferred DataCenter
+/// Accepts the DC name of the ScyllaDB to filter the connection to that DC only (preferrably).
+/// If you connect to multi-DC cluter, you might experience big latencies while working with the DB.
+/// This is due to the fact that ScyllaDB driver tries to connect to any of the nodes in the cluster
+/// disregarding of the location of the DC. This option allows to filter the connection to the DC you need.
+/// Example: "DC1" where DC1 is located in the same region as the application.
+///
+/// `keepalive_interval: Option<u64>`
+/// ScyllaDB keepalive interval
+///
+/// `max_retry: u8`
+/// Max retry count for ScyllaDB if `strict_mode` is `false`
+///
+/// `strict_mode: bool`
+/// Attempts to store data in the database should be infinite to ensure no data is missing.
+/// Disable it to perform a limited write attempts (`max_retry`)
+/// before skipping giving up and moving to the next piece of data
+#[cfg(feature = "scylla_db")]
+pub struct AdditionalDatabaseOptions {
```

Comment:
I'd suggest moving this definition to `scylladb` mod so it it imported with a mod instead of defining it here under the feature flag